UnitTests/PaymentsCoreTests/APIClient_Tests.swift

```diff
@@ -73,107 +116,94 @@ class APIClient_Tests: XCTestCase {
         waitForExpectations(timeout: 1)
     }
 
-    func testFetch_withAccessTokenInvalidMockResponse_vendsDecodingError() {
+    func testFetch_whenInvalidData_returnsParseError() {
         let expect = expectation(description: "Get mock response for access token request")
 
-        let mockInvalidResponse: String = """
+        let jsonResponse = """
         {
             "test": "wrong response format"
         }
         """
 
-        let mockRequestResponse = RequestResponseMock(
-            request: AccessTokenRequest(clientID: ""),
-            statusCode: 200,
-            responseString: mockInvalidResponse
-        )
-
-        URLProtocolMock.requestResponses.append(mockRequestResponse)
+        mockURLSession.cannedURLResponse = successURLResponse
+        mockURLSession.cannedData = jsonResponse.data(using: String.Encoding.utf8)!
 
-        apiClient.fetch(endpoint: mockRequestResponse) { result, _ in
-            guard case .failure(.parsingError) = result else {
-                XCTFail("Expect NetworkingError.decodingError for invalid response")
-                return
+        apiClient.fetch(endpoint: fakeRequest) { result, _ in
+            switch result {
+            case .success(_):
+                XCTFail()
+            case .failure(let error):
+                XCTAssertNotNil(error)
             }
 
             expect.fulfill()
         }
         waitForExpectations(timeout: 1)
     }
 
-    func testFetch_withRequestExpectingEmptyResponse_vendsSuccessResult() {
-        let expect = expectation(description: "Get empty response type for mock request")
-        let emptyRequest = EmptyRequestResponseMock()
+    // TODO: Get more granual here. Also, do we want to move this check for
+    // non-success status code above data parsing in our source code?
+    func testFetch_whenBadStatusCode_returnsUnknownError() {
+        let expect = expectation(description: "Get mock response for access token request")
 
-        URLProtocolMock.requestResponses.append(emptyRequest)
+        let jsonResponse = """
+        { "some": "json" }
+        """
 
-        apiClient.fetch(endpoint: emptyRequest) { result, _ in
-            guard case .success = result else {
-                XCTFail("Expected successful empty response")
-                return
-            }
-            expect.fulfill()
-        }
-        waitForExpectations(timeout: 1)
-    }
+        mockURLSession.cannedData = jsonResponse.data(using: String.Encoding.utf8)!
```

Comment:
we can avoid force unwrapping in tests is to make the func throwing and then use `XCTUnwrap` -
```swift
func testFetch_whenBadStatusCode_returnsUnknownError() throws {
...
mockURLSession.cannedData = try XCTUnwrap(jsonResponse.data(using: String.Encoding.utf8))
...
```
Not that this would ever realistically fail, but a crash on a force unwrap would abort the test suite from running rather than fail the test, so as we get a lot more tests / code to test, defaulting to this would be a good way to avoid unwrapping and minimize that risk

Reply:
Gotcha - yeh that's a great callout. There are a few other comments on this PR to move the string -> data conversion anyways, so I'll address the force unwrap in that slight refactor.